10up/wp-scaffold
# Enhance css reset

### Description of the Change
This pull request adds a new CSS file named `reset.css`. This file is used to enhance our existing reset.

Some file structure names were altered and names changed for the primary files in `frontend.css`

- Numbered the order of files - for more clear hierarchy and compilation sequence
- Added a `01-Settings` directory (including `index.css`, `colors,css`, `custom-selectors.css`, and `media-queries.css` 
- The previously mentioned `custom-selectors.css` file was added to enable bulk selectors
- In `02-Global`, `index.css`, `mixins.css`, and the new `reset.css` were added.

Styles added to the `reset.css` file:

- `font: inherit` globally to cover for some browser inconsistencies and provide more font display reliability
-- NOTICE: this will flatten all browser font styles and rely 100% on styles coming from the site
- `margin: 0; padding: 0;` globally to ensure there is a reliable spacing reset
- `color-scheme: dark light;` to html - add default color schemes to project - available if needed
- `hanging-punctuation: first last;` - enhance block quotes or large content blocks in quotes
- `body { min-height: 100svh; }` ensure body is at least 100% of viewport hight (including on mobile devices)
- Responsive media: set to `display: block;` and `max-width: 100%;` by default
- All headings: add `text-wrap: balance` allow browser to set a balanced heading if more than one line
- All paragraphs: add `text-wrap: pretty` ensure there are no orphans on paragraphs that are more than one line
- Add smooth scrolling by default 

These styles were tested locally and work as expected.

Closes #210

### How to test the Change
Once styles are loaded into a project, you can test each enhancement individually.

Assuming no custom styles added
1. All text will be the same font (style, weight, etc)
2. No margin or padding will exist
3. You can add `@media (prefers-color-scheme: [dark or light]) {}` media queries to test styles based on chosen color scheme
4. You can add a large paragraph wrapped in quotes, and the quotes to hang (text vertically aligned left and right) by default
5. If you select the body element - it will grow to fit the entire viewport height - even on mobile devices
6. This should be basic
7. Headings should evenly balance across multiple lines (on supporting browsers)
8. Paragraphs should not leave an orphan word when spanning multiple lines (on supporting browsers)
9. Smooth scrolling will animate the browser scrolling when a user clicks an element linking to a different section on the same page. This will also work within modals.

### Changelog Entry

> Added - Custom CSS Reset enhancements, new CSS folder, `custom-selectors.css`
> Changed - Existing CSS directories in `frontend`


### Credits


### Checklist:
